src/lib/dlt_user.c

```diff
@@ -4761,6 +4844,79 @@ DltReturnValue dlt_user_log_check_user_message(void)
                         return DLT_RETURN_ERROR;
                 }
                 break;
+#ifdef DLT_TRACE_LOAD_CTRL_ENABLE
+                case DLT_USER_MESSAGE_TRACE_LOAD:
+                {
+                    /*
+                     * at least user header and message length is available
+                     */
+                    trace_load_settings_user_message_bytes_required =
+                        (int32_t) (sizeof(DltUserHeader) + sizeof(uint32_t ));
+                    if (receiver->bytesRcvd < (int32_t)trace_load_settings_user_message_bytes_required) {
+                        // Not enough data to read the message length
+                        leave_while = 1;
+                        break;
+                    }
+
+                    // Read trace settings count from buffer.
+                    trace_load_settings_user_messages_count = (uint32_t)(*(receiver->buf + sizeof(DltUserHeader)));
+                    trace_load_settings_user_message_bytes_required +=
+                        trace_load_settings_user_messages_count * sizeof(DltUserControlMsgTraceSettingMsg);
+                    if (receiver->bytesRcvd < (int32_t)trace_load_settings_user_message_bytes_required) {
+                        // Not enough data to read trace settings
+                        leave_while = 1;
+                        break;
+                    }
+
+                    trace_load_settings_user_messages =
+                        (DltUserControlMsgTraceSettingMsg *)(receiver->buf + sizeof(DltUserHeader) + sizeof(uint32_t));
+
+                    pthread_rwlock_wrlock(&trace_load_rw_lock);
+
+                    // Remove the default created at startup
+                    if (trace_load_settings != NULL) {
+                        free(trace_load_settings);
+                    }
+
+                    char msg[255];
+                    trace_load_settings_alloc_size = sizeof(DltTraceLoadSettings) * trace_load_settings_user_messages_count;
+                    trace_load_settings = malloc(trace_load_settings_alloc_size);
```

Comment:
Kindly please ensure that the memory is allocated successfully before accessing.

Reply:
done